src/topology/path_connected.lean

```diff
@@ -469,6 +714,69 @@ begin
   exact ⟨(hx hyW).joined_subtype.some_path.map (continuous_inclusion hWU), by simp⟩
 end
 
+lemma is_path_connected.exists_path_through_family
+  {X : Type*} [topological_space X] {n : ℕ} {s : set X} (h : is_path_connected s)
+  (p : fin (n+1) → X) (hp : ∀ i, p i ∈ s) : ∃ γ : path (p 0) (p n), (∀ i, p i ∈ range γ) ∧ (range γ ⊆ s) :=
+begin
+  let p' : ℕ → X := λ k, if h : k < n+1 then p ⟨k, h⟩ else p ⟨0, n.zero_lt_succ⟩,
+  obtain ⟨γ, hγ⟩ : ∃ (γ : path (p' 0) (p' n)), (∀ i ≤ n, p' i ∈ range γ) ∧ range γ ⊆ s,
+  { have hp' : ∀ i ≤ n, p' i ∈ s,
+    { intros i hi,
+      simp [p', nat.lt_succ_of_le hi, hp] },
+    clear_value p',
+    clear hp p,
+    induction n with n hn,
+    { use (λ _, p' 0),
+      { continuity },
+      { split,
+        { rintros i hi, rw nat.le_zero_iff.mp hi, exact ⟨0, rfl⟩ },
+        { rw range_subset_iff, rintros x, exact hp' 0 (le_refl _) } } },
+    { rcases hn (λ i hi, hp' i $ nat.le_succ_of_le hi) with ⟨γ₀, hγ₀⟩,
+      rcases h.joined_in (p' n) (p' $ n+1) (hp' n n.le_succ) (hp' (n+1) $ le_refl _) with ⟨γ₁, hγ₁⟩,
+      let γ : path (p' 0) (p' $ n+1) := γ₀.trans γ₁,
+      use γ,
+      have range_eq : range γ = range γ₀ ∪ range γ₁ := γ₀.trans_range γ₁,
+      split,
+      { rintros i hi,
+        by_cases hi' : i ≤ n,
+        { rw range_eq,
+          left,
+          exact hγ₀.1 i hi' },
+        { rw [not_le, ← nat.succ_le_iff] at hi',
+          have : i = n.succ := by linarith,
+          rw this,
+          use 1,
+          exact γ.target } },
+      { rw range_eq,
+        apply union_subset hγ₀.2,
+        rw range_subset_iff,
+        exact hγ₁ } } },
+  have hpp' : ∀ k < n+1, p k = p' k,
+  { intros k hk, simp only [p', hk, dif_pos], congr, ext, rw fin.coe_coe_of_lt hk, norm_cast },
+  use γ.cast (hpp' 0 n.zero_lt_succ) (hpp' n n.lt_succ_self),
+  simp only [γ.cast_coe],
+  refine and.intro _ hγ.2,
+  rintros ⟨i, hi⟩,
+  convert hγ.1 i (nat.le_of_lt_succ hi), rw ← hpp' i hi,
+  congr,
+  ext,
+  rw fin.coe_coe_of_lt hi,
+  norm_cast
+end
+
+lemma is_path_connected.exists_path_through_family'
+  {X : Type*} [topological_space X] {n : ℕ} {s : set X} (h : is_path_connected s)
+  (p : fin (n+1) → X) (hp : ∀ i, p i ∈ s) :
+  ∃ (γ : path (p 0) (p n)) (t : fin (n + 1) → I), (∀ t, γ t ∈ s) ∧ ∀ i, γ (t i) = p i :=
```

Comment:
You have exchanged the order of the conclusion compared to the previous lemma (here, range inside `s` comes first, and then the fact that the `p i` are realized; there, the opposite). Can you make it coherent?